packages/protocol-kit/src/SafeProvider.ts

```diff
@@ -32,49 +22,114 @@ import {
   SafeProviderConfig,
   Eip1193Provider,
   HttpTransport,
-  SocketTransport
+  SocketTransport,
+  ExternalSigner
 } from '@safe-global/protocol-kit/types'
+import { asAddress, asHash, asHex, getChainById } from './utils/types'
+import {
+  createPublicClient,
+  createWalletClient,
+  PublicClient,
+  custom,
+  http,
+  getAddress,
+  isAddress,
+  Transaction,
+  decodeAbiParameters,
+  encodeAbiParameters,
+  parseAbiParameters,
+  toBytes,
+  BlockTag,
+  Chain
+} from 'viem'
+import { privateKeyToAccount } from 'viem/accounts'
+import {
+  toEstimateGasParameters,
+  toCallGasParameters,
+  sameString
+} from '@safe-global/protocol-kit/utils'
+
+function asBlockId(blockId: number | string | undefined) {
```

Comment:
Shouldn't be this utils in an utils file?

Reply:
Yes, done!